<a href="https://colab.research.google.com/github/G3ntl3g1ant/Coffee-Analysis/blob/main/Project_2_Part_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Imports

In [2]:
# Imports
import pandas as pd
import numpy as np
import seaborn as sns

# Data Sets Considered for Processing

## Coffee Reviews Dataset

In [3]:
path = 'https://docs.google.com/spreadsheets/d/e/2PACX-1vRa24vFIf2Nb9LZQcARBewLWJAq7RkLNyTGwmcUxF6gG3zieLxBQGx5a7n5e_YxDgbtEqAClInFZwim/pub?gid=2080147348&single=true&output=csv'
df = pd.read_csv(path)
df.head()

name                    roaster         roast  \
0     “Sweety” Espresso Blend                     A.R.C.  Medium-Light   
1        Flora Blend Espresso                     A.R.C.  Medium-Light   
2    Ethiopia Shakiso Mormora               Revel Coffee  Medium-Light   
3          Ethiopia Suke Quto                Roast House  Medium-Light   
4  Ethiopia Gedeb Halo Beriti  Big Creek Coffee Roasters        Medium   

     loc_country        origin_1           origin_2  100g_USD  rating  \
0      Hong Kong          Panama           Ethiopia     14.32      95   
1      Hong Kong          Africa       Asia Pacific      9.05      94   
2  United States       Guji Zone  Southern Ethiopia      4.70      92   
3  United States       Guji Zone      Oromia Region      4.19      92   
4  United States  Gedeb District         Gedeo Zone      4.85      94   

     review_date                                             desc_1  \
0  November 2017  Evaluated as espresso. Sweet-toned, deeply ric...   
1  November 2017  Evaluated as espresso. Sweetly tart, floral-to...   
2  November 2017  Crisply sweet, cocoa-toned. Lemon blossom, roa...   
3  November 2017  Delicate, sweetly spice-toned. Pink peppercorn...   
4  November 2017  Deeply sweet, subtly pungent. Honey, pear, tan...   

                                              desc_2  \
0  An espresso blend comprised of coffees from Pa...   
1  An espresso blend comprised of coffees from Af...   
2  This coffee tied for the third-highest rating ...   
3  This coffee tied for the third-highest rating ...   
4  Southern Ethiopia coffees like this one are pr...   

                                              desc_3  
0  A radiant espresso blend that shines equally i...  
1  A floral-driven straight shot, amplified with ...  
2  A gently spice-toned, floral- driven wet-proce...  
3  Lavender-like flowers and hints of zesty pink ...  
4  A deeply and generously lush cup saved from se...

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2095 entries, 0 to 2094
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   name         2095 non-null   object 
 1   roaster      2095 non-null   object 
 2   roast        2080 non-null   object 
 3   loc_country  2095 non-null   object 
 4   origin_1     2095 non-null   object 
 5   origin_2     2095 non-null   object 
 6   100g_USD     2095 non-null   float64
 7   rating       2095 non-null   int64  
 8   review_date  2095 non-null   object 
 9   desc_1       2095 non-null   object 
 10  desc_2       2095 non-null   object 
 11  desc_3       2093 non-null   object 
dtypes: float64(1), int64(1), object(10)
memory usage: 196.5+ KB


**Source of the Data:**

https://www.kaggle.com/datasets/schmoyote/coffee-reviews-dataset

**Data Description:**
Coffee Reviews comparing bean roast, roaster, growth location and price per 100 grams, etc.

**Target:** Rating of a coffee

This will be the target because we are trying to find the best/most popular coffee brand/roast. 


**Each Row Represents:** 1 person/reviewer

**Classification or Regression:** Classification

We are looking to find out the approval ratings of present and new coffees. 

**Number of Features:** 7 Features 

(The other present features will be removed during the cleaning phase)

**Number of Rows:** 2095 Rows

**Challenges:** With this Dataset I can forsee some challenges with the origin columns, how they will influence the Dataset, I cannot as of yet say. 

But modeling for sure will be a challenge with so many object values. Will have to be careful to categorize the objects correctly, wheather they are ordinal or nominal, respectively. 

## Starbucks Customer Survey Dataset

In [5]:
path2 = 'https://docs.google.com/spreadsheets/d/e/2PACX-1vTM74mIJ9C-gx3KWHUDNFBIMAZRHcyRm0ZtZAj30NcAHjmsfJkrtAfIakreE9Ec1_avOOGVRBDat0E2/pub?gid=2132854527&single=true&output=csv'
df2 = pd.read_csv(path2)
df2.head()

Timestamp 1. Your Gender    2. Your Age  \
0  2019/10/01 12:38:43 PM GMT+8         Female  From 20 to 29   
1  2019/10/01 12:38:54 PM GMT+8         Female  From 20 to 29   
2  2019/10/01 12:38:56 PM GMT+8           Male  From 20 to 29   
3  2019/10/01 12:39:08 PM GMT+8         Female  From 20 to 29   
4  2019/10/01 12:39:20 PM GMT+8           Male  From 20 to 29   

  3. Are you currently....? 4. What is your annual income?  \
0                   Student             Less than RM25,000   
1                   Student             Less than RM25,000   
2                  Employed             Less than RM25,000   
3                   Student             Less than RM25,000   
4                   Student             Less than RM25,000   

  5. How often do you visit Starbucks? 6. How do you usually enjoy Starbucks?  \
0                               Rarely                                Dine in   
1                               Rarely                              Take away   
2                              Monthly                                Dine in   
3                               Rarely                              Take away   
4                              Monthly                              Take away   

  7. How much time do you normally  spend during your visit?  \
0                       Between 30 minutes to 1 hour           
1                                   Below 30 minutes           
2                       Between 30 minutes to 1 hour           
3                                   Below 30 minutes           
4                       Between 30 minutes to 1 hour           

  8. The nearest Starbucks's outlet to you is...?  \
0                                      within 1km   
1                                       1km - 3km   
2                                   more than 3km   
3                                   more than 3km   
4                                       1km - 3km   

  9. Do you have Starbucks membership card?  ...  \
0                                       Yes  ...   
1                                       Yes  ...   
2                                       Yes  ...   
3                                       Yes  ...   
4                                       Yes  ...   

  11. On average, how much would you spend at Starbucks per visit?  \
0                                     Less than RM20                 
1                                     Less than RM20                 
2                                     Less than RM20                 
3                                     Less than RM20                 
4                                 Around RM20 - RM40                 

  12. How would you rate the quality of Starbucks compared to other brands (Coffee Bean, Old Town White Coffee..) to be:  \
0                                                  4                                                                       
1                                                  4                                                                       
2                                                  4                                                                       
3                                                  2                                                                       
4                                                  3                                                                       

   13. How would you rate the price range at Starbucks?  \
0                                                  3      
1                                                  3      
2                                                  3      
3                                                  1      
4                                                  3      

   14. How important are sales and promotions in your purchase decision?  \
0                                                  5                       
1                                                  4                       
2    

In [6]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 121 entries, 0 to 120
Data columns (total 21 columns):
 #   Column                                                                                                                  Non-Null Count  Dtype 
---  ------                                                                                                                  --------------  ----- 
 0   Timestamp                                                                                                               121 non-null    object
 1   1. Your Gender                                                                                                          121 non-null    object
 2   2. Your Age                                                                                                             121 non-null    object
 3   3. Are you currently....?                                                                                               121 non-null    object
 4   4.

**Source of the Data:**

https://www.kaggle.com/datasets/malaynarashid/starbucks-customer-survey

**Description:** Survey results for Starbucks Customers

**Target:** Membership

**Each Row Represents:** 1 Individual

**Classification or Regression:** Classification

**Number of Features:** 17 Features

**Number of Rows:** 121

**Challenges:** First and foremost would be renaming the columns to clean up the size of the labels to begin with will be a big challenge. Plus taking the data provided and correctly changing the ratings as well as the objects into the proper variables to preprocess the whole dataframe. 

Though most of this challenge will be in the cleaning and the pipeing into the model correctly. 